cmd/app.go

```diff
@@ -0,0 +1,188 @@
+	if err != nil {
+		log.Fatalf("Failed to connect: %v\n", err)
+	}
+	defer conn.Close()
+
+	//restaurant
+	repo := repository.NewRestaurantRepo(db)
+	ru := serviceR.NewRestaurantUsecace(repo)
+	rs := handlers.NewRestaurantService(ru, cust.NewOfficeServiceClient(conn), cust.NewUserServiceClient(conn))
+
+	//customer
+	repoC := cusrepository.NewCustomerRepo(gorm)
+	cu := service.NewCustomerUsecase(repoC)
+	cs := handlerscustomer.New(rest.NewMenuServiceClient(conn), cu)
+
+	//statistic
+	repoS := statrepository.NewStatRepo(dbx)
+	su := statservice.NewStatUsecase(repoS)
+	sh := stathandlers.NewStatService(rest.NewProductServiceClient(conn), su)
+
+	s := grpc.NewServer()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	rest.RegisterProductServiceServer(s, &handlers.RestaurantService{})
+	rest.RegisterMenuServiceServer(s, &handlers.RestaurantService{})
+	rest.RegisterOrderServiceServer(s, &handlers.RestaurantService{})
+	cust.RegisterOrderServiceServer(s, &handlerscustomer.CustomerService{})
+	cust.RegisterOfficeServiceServer(s, &handlerscustomer.CustomerService{})
+	cust.RegisterUserServiceServer(s, &handlerscustomer.CustomerService{})
+	statistics.RegisterStatisticsServiceServer(s, &stathandlers.StatisticService{})
+
+	router := runtime.NewServeMux()
+
+	err = rest.RegisterProductServiceHandlerServer(ctx, router, rs)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+	err = rest.RegisterMenuServiceHandlerServer(ctx, router, rs)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+
+	err = rest.RegisterOrderServiceHandlerServer(ctx, router, rs)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+
+	err = cust.RegisterOrderServiceHandlerServer(ctx, router, cs)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+
+	err = cust.RegisterOfficeServiceHandlerServer(ctx, router, cs)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+
+	err = cust.RegisterUserServiceHandlerServer(ctx, router, cs)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+
+	err = statistics.RegisterStatisticsServiceHandlerServer(ctx, router, sh)
+	if err != nil {
+		log.Printf("Failed to register gateway: %v\n", err)
+	}
+
+	log.Printf("Starting GRPCRestaurantServer at port: %v\n ", conf.API.Host)
+	return http.ListenAndServe(conf.API.Host, httpGrpcRouter(s, router))
+
+}
+
+// TODO: проверить как работает без этого роутера, может из-за него не получается выполнить запрос в рамках одного сервера
+func httpGrpcRouter(grpcServer *grpc.Server, httpHandler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
+			grpcServer.ServeHTTP(w, r)
+		} else {
+			httpHandler.ServeHTTP(w, r)
+		}
+	})
+}
+
+func StartGRPC(conf configs.Config) {
```

Comment:
Если в сигнатуру добавить error, вот так:

```suggestion
func StartGRPC(conf configs.Config) (err error) {
```
то все последующие 
```go
log.Fatal("cannot connect to db: ", err)
```

можно будет заменить на:
```go
return err
```